graylog2-server/src/main/java/org/graylog2/dashboards/DashboardImpl.java

```diff
@@ -72,6 +79,42 @@ public void setDescription(String description) {
         this.fields.put(FIELD_DESCRIPTION, description);
     }
 
+    @Override
+    public List<WidgetPosition> getPositions() {
+        final BasicDBObject positions = (BasicDBObject) fields.get(DashboardImpl.EMBEDDED_POSITIONS);
+        if (positions == null) {
+            return Collections.emptyList();
+        }
+
+        final List<WidgetPosition> result = new ArrayList<>(positions.size());
+        for ( String positionId : positions.keySet() ) {
+            final BasicDBObject position = (BasicDBObject) positions.get(positionId);
+            final Integer width = parseInt(position.get("width").toString());
+            final Integer height = parseInt(position.get("height").toString());
+            final Integer col = parseInt(position.get("col").toString());
+            final Integer row = parseInt(position.get("row").toString());
+            result.add(WidgetPosition.create(positionId, width, height, col, row));
+        }
+        return result;
+    }
+
+    @Override
+    public void setPositions(List<WidgetPosition> widgetPositions) {
+        if (widgetPositions == null) {
+            return;
+        }
+        final Map<String, Map<String, Object>> positions = new HashMap<>(widgetPositions.size());
+        for (WidgetPosition widgetPosition : widgetPositions) {
+            Map<String, Object> position = new HashMap<>(4);
+            position.put("width", widgetPosition.width());
+            position.put("height", widgetPosition.height());
+            position.put("col", widgetPosition.col());
+            position.put("row", widgetPosition.row());
+            positions.put(widgetPosition.id(), position);
```

Comment:
While this would work, it is unnecessarily complex and error-prone. What would happen if an attribute is added to the `WidgetPosition` class, but forgotten to transfer it in here?

My suggestion is this: Make the `WidgetPosition` class serializable to JSON by adding `@JsonProperty` annotations to all getters (and `@JsonIgnore` `id()`) and just put the value class in the dashboard field.

Reply:
After a short discussion with @bernd we came to the conclusion that for now it would be too much work to use the ```JsonProperty``` here since the underlying ```PersistendImpl``` cannot use it.

Reply:
:+1: